apps/backend/src/modules/users/user.repository.ts

```diff
@@ -41,10 +45,18 @@ class UserRepository implements Repository {
 		return user ? UserEntity.initialize(user) : null;
 	}
 
-	public async findAll(): Promise<UserEntity[]> {
-		const users = await this.userModel.query().execute();
+	public async findAll({
+		page,
+		pageSize,
+	}: PaginationQueryParameters): Promise<PaginationResponseDto<UserEntity>> {
+		const { results, total } = await this.userModel
+			.query()
+			.page(--page, pageSize);
```

Comment:
this should be done on service level. Also use `- PAGE_INDEX_OFFSET` instead of mutating parameters.
```suggestion
			.page(page, pageSize);
```